api/v1alpha08/kogitoserverlessplatform_types.go

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha08
+
+import (
+	"github.com/kiegroup/container-builder/api"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ConfigurationSpec represents a generic configuration specification
+type ConfigurationSpec struct {
+	// Type represents the type of configuration, ie: property, configmap, secret, ...
+	Type string `json:"type"`
+	// Value the value to assign to the configuration (syntax may vary depending on the `Type`)
+	Value string `json:"value"`
+}
+
+const (
+	// KogitoServerlessPlatformKind is the Kind name of the KogitoServerlessPlatform CR
+	KogitoServerlessPlatformKind string = "KogitoServerlessPlatform"
+)
+
+// PlatformCluster is the kind of orchestration cluster the platform is installed into
+type PlatformCluster string
+
+const (
+	// PlatformClusterOpenShift is used when targeting an OpenShift cluster
+	PlatformClusterOpenShift PlatformCluster = "OpenShift"
+	// PlatformClusterKubernetes is used when targeting a Kubernetes cluster
+	PlatformClusterKubernetes PlatformCluster = "Kubernetes"
+)
+
+// PlatformBuildStrategy specifies how the Build should be executed.
+// It will trigger a Kaniko pod that will take care of producing the expected image containing the Quarkus/Kogito runtime.
+// +kubebuilder:validation:Enum=pod
+type PlatformBuildStrategy string
+
+const (
+	// BuildStrategyPod performs the build in a `Pod` (will schedule a new builder ephemeral `Pod` which will take care of the build action).
+	BuildStrategyPod PlatformBuildStrategy = "pod"
+)
+
+// PlatformBuildStrategies is a list of strategies allowed for the build
+var PlatformBuildStrategies = []PlatformBuildStrategy{
```

Comment:
we can add in the future the possibility to run a image that already is built and available, for example, on quay.

Reply:
yes this struct is created for this purpose.